### PR TITLE
Black 380: action progress 

### PR DESCRIPTION
**Pull request by stabacco**
Adds a progress bar to have a clearer idea about the execution.

### CODE REVIEW

File: qctrl/optimizations.py [hunk 1]
```diff
@@ -97,10 +99,23 @@ def _run(self, system: resourceobject.ResourceObject,
         # return a fresh copy of the system, with the optimization values
         # saved in.
         action.session.invalidate()
-        return self.api.systems.find(system.id)
+        return self.api.systems.find_by_id(system.id)
```
Comment (saltysealion) [inline qctrl/optimizations.py]: COOL @stabacco

File: qctrl/optimizations.py [hunk 2]
```diff
@@ -252,7 +282,8 @@ def _new(self, data: dict) -> resourceobject.ResourceObject:
                 pulse.fields["type"] = pulse_type
                 pulse.segment_count = segment_count
                 pulse.upper_bound = serialized_control["upper_bound"]
-                pulse.fixed_modulus = serialized_control["fixed_modulus"] if pulse_type == "complex" else False
+                pulse.fixed_modulus = serialized_control["fixed_modulus"] \
+                    if pulse_type == "complex" else False
```
Comment (saltysealion) [inline qctrl/optimizations.py]: Should `complex` be a constant somewhere?

File: qctrl/qctrl.py [hunk 3]
```diff
@@ -65,7 +64,7 @@ def find(self, system_id: Union[str, int]) -> \
         :param str/int system_id: the id of the `System` resource.
         :return resourceobject.ResourceObject: the `System` resource object.
         """
-        return self.api.systems.find(system_id)
+        return self.api.systems.find(system_id)  # @pylint: disable=no-member
```
Comment (saltysealion) [inline qctrl/qctrl.py]: More of a style, but I group pylint at the top, typically under a class definition

File: qctrl/schema.py [hunk 4]
```diff
@@ -89,7 +89,7 @@ class ControlSchema(Schema):
     pulse = fields.Nested(PulseSchema, required=False, load_only=True)
 
     @pre_load
-    def validate(self, data):
+    def validate_control(self, data):
```
Comment (saltysealion) [inline qctrl/schema.py]: Do you need to specify that it's a control here given that it is a control schema?

File: qctrl/optimizations.py (hunk 2 above)
Comment (stabacco) [reply qctrl/optimizations.py]: Yes, it probably should be defined in the commons somewhere. I'll add it in the next release.

**Pull request merged**